showcase/src/app/demo/toast/toast-notification.component.ts

```diff
@@ -0,0 +1,119 @@
+	 */
+	public hideDisabled: boolean = true;
+
+	/**
+	 * Class constructor
+	 * @param logger - The logger of the application
+	 * @param toastService - The toast notification service
+	 * @param ref - Change detector used to refresh the angular bindings
+	 */
+	public constructor(
+		@Inject(STARK_LOGGING_SERVICE) public logger: StarkLoggingService,
+		@Inject(STARK_TOAST_NOTIFICATION_SERVICE) public toastService: StarkToastNotificationService,
+		public ref: ChangeDetectorRef
+	) {}
+
+	/**
+	 * Opens an error toast
+	 * @param message - Message displayed in the toast
+	 * @param delay - Delay before toast hides (0 to avoid automatic hiding)
+	 * @param action - Label of the action button (empty to remove the action button)
+	 */
+	public openErrorToast(message: string, delay: number = 3000, action?: string): void {
+		this.openToast(message, delay, action, StarkMessageType.ERROR);
+	}
+
+	/**
+	 * Opens an warning toast
+	 * @param message - Message displayed in the toast
+	 * @param delay - Delay before toast hides (0 to avoid automatic hiding)
+	 * @param action - Label of the action button (empty to remove the action button)
+	 */
+	public openWarningToast(message: string, delay: number = 3000, action?: string): void {
+		this.openToast(message, delay, action, StarkMessageType.WARNING);
+	}
+
+	/**
+	 * Opens a toast
+	 * @param message - Message displayed in the toast
+	 * @param delay - Delay before toast hides (0 to avoid automatic hiding)
+	 * @param action - Label of the action button (empty to remove the action button)
+	 * @param type - Type of toast
+	 */
+	public openToast(message: string, delay?: number, action?: string, type: StarkMessageType = StarkMessageType.INFO): void {
+		this.hideDisabled = false;
+		const toastData: StarkToastNotificationMessage = {
+			key: message,
+			id: "1",
+			code: "abc",
+			type: type,
+			actionLabel: action
+		};
+		if (typeof delay !== "undefined") {
+			toastData.delay = delay;
+		}
+		this.toastService.show(toastData).subscribe((result: StarkToastNotificationResult) => {
+			/** result contains the type of message sent by the toast service */
+			switch (result) {
+				case StarkToastNotificationResult.CLOSED_ON_DELAY_TIMEOUT:
+					this.messageText = "SHOWCASE.DEMO.TOAST.MESSAGE_CLOSED_ON_DELAY_TIMEOUT";
+					this.hideDisabled = true;
+					break;
+				case StarkToastNotificationResult.ACTION_CLICKED:
+					this.messageText = "SHOWCASE.DEMO.TOAST.MESSAGE_ACTION_CLICKED";
+					this.hideDisabled = true;
+					break;
+				case StarkToastNotificationResult.HIDDEN:
+					this.messageText = "SHOWCASE.DEMO.TOAST.MESSAGE_HIDDEN";
+					this.hideDisabled = true;
+					break;
+				case StarkToastNotificationResult.CLOSED_BY_NEW_TOAST:
+					this.messageText = "SHOWCASE.DEMO.TOAST.MESSAGE_CLOSED_BY_NEW_TOAST";
+					break;
+				default:
+					this.messageText = "";
+					this.hideDisabled = true;
+					throw new Error("Unknown toast notification result!!");
+			}
+			this.ref.detectChanges();
```

Comment:
Why using `ChangeDetectorRef` here ? Cannot we use toast-notification without using this ?
In table component, we use it ourselves in stark-ui/table module..

I think it's good to force the developer to use this 😕